src/pkg/cluster/tunnel.go

```diff
@@ -153,6 +154,15 @@ func (c *Cluster) ConnectToZarfRegistryEndpoint(ctx context.Context, registryInf
 		if tunnel, err = c.NewTunnel(ZarfNamespaceName, SvcResource, ZarfRegistryName, "", 0, ZarfRegistryPort); err != nil {
 			return "", tunnel, err
 		}
+	} else if dns.IsServiceURL(registryInfo.Address) {
```

Comment:
In the packager2 mirror refactor the push logic was changed to take Kubernetes services as input, this brings this function in parity with that.